syntax = "proto3";

package proto;

import "common.proto";

option go_package = "app/generated/go/proto";

message Vector {
  repeated float entry = 1;
}

message GetClosestNeighborRequest {
  optional Vector vector = 1;
}
message GetClosestNeighborResponse {
  optional string author = 1;
}

message FindAverageVectorRequest {}
message FindAverageVectorResponse {
  optional Vector vector = 1;
}

service SimilarityEngineService {
  rpc GetClosestNeighbor(GetClosestNeighborRequest)
      returns (GetClosestNeighborResponse);
  rpc FindAverageVector(FindAverageVectorRequest)
      returns (FindAverageVectorResponse);
  rpc CheckHealth(HealthCheckRequest) returns (Payload);
}